thread.c

```diff
@@ -545,9 +545,125 @@ static void thread_libevent_process(int fd, short which, void *arg) {
     }
 }
 
-/* Which thread we assigned a connection to most recently. */
+/* The thread we assigned a connection to most recently. */
 static int last_thread = -1;
 
+/* This function simply returns next_thread based on round-robin */
+static LIBEVENT_THREAD *select_next_thread(void)
+{
+    int tid = (last_thread + 1) % settings.num_threads;
+    last_thread = tid;
+    return threads + tid;
+}
+
+/* This function selects a thread to process requests from a given connection.
+ * Thread selection is either based on the (existing) default mechanism,
+ * round-robin based thread selection, or based on NAPI_ID. NAPI_ID based
+ * selection creates an association between a (hardware) queue on a NIC and an
+ * application thread. This mapping streamlines the flow of data from the NIC
+ * to an application thread by having an application thread source data from
+ * the same hardware queue which serves up traffic for many connections
+ */
+static void select_thread(int sfd, LIBEVENT_THREAD **thread_selected)
```

Comment:
Why does `select_next_thread` return a LIBEVENT_THREAD, but `select_thread` is void with an input pointer?

Reply:
I can make that change, Absolutely no problem.